codec/src/main/java/io/netty/handler/codec/compression/SnappyFrameEncoder.java

```diff
@@ -120,4 +125,29 @@ private static void writeChunkLength(ByteBuf out, int chunkLength) {
     private static void calculateAndWriteChecksum(ByteBuf slice, ByteBuf out) {
         out.writeIntLE(calculateChecksum(slice));
     }
+
+    @Override
+    public boolean isClosed() {
+        return false;
+    }
+
+    @Override
+    public ChannelFuture close() {
+        return ctx().newFailedFuture(new UnsupportedOperationException());
+    }
+
+    @Override
+    public ChannelFuture close(ChannelPromise promise) {
+        return promise.setFailure(new UnsupportedOperationException());
+    }
+
+    @Override
+    public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exception {
+        ctx.close(promise);
+    }
+
+    @Override
+    protected ChannelFuture finishEncode(ChannelHandlerContext ctx, ChannelPromise promise) {
+        return promise.setFailure(new UnsupportedOperationException());
```

Comment:
Instead of rejecting the close request, we could simply close the channel immediately without scheduling any write timeout, etc?